Propagate the javax.annotation.security annotations in REST Data

Sgitario opened the pull request:
With these changes, the REST Data with Panache extension will propagate the Security annotations within the package `javax.annotation.security` that are defined on your resource interfaces:

```java
import javax.annotation.security.DenyAll;
import javax.annotation.security.RolesAllowed;

@DenyAll
@ResourceProperties
public interface PeopleResource extends PanacheEntityResource<Person, Long> {
    @RolesAllowed("superuser")
    boolean delete(Long id);
}
```

Additionally, if you are only interested in specifying the roles that are allowed to use the resources, the `@ResourceProperties` and `@MethodProperties` annotations have the field `rolesAllowed` to list the security roles permitted to access the resource or operation.

Fix https://github.com/quarkusio/quarkus/issues/28995

Sgitario commented:
cc @geoand @punkratz312

sberyozkin commented:
@Sgitario Hi Jose, can you please add a small section to the main `security.adoc` to link back to the rest-panache doc, for example, just next to https://quarkus.io/guides/security#secure-serialization

gsmet commented on extensions/panache/rest-data-panache/deployment/src/main/java/io/quarkus/rest/data/panache/deployment/properties/ResourcePropertiesProvider.java:
```diff
@@ -19,6 +23,8 @@ public class ResourcePropertiesProvider {
     private static final DotName METHOD_PROPERTIES_ANNOTATION = DotName.createSimple(
             io.quarkus.rest.data.panache.MethodProperties.class.getName());
 
+    private static final List<String> ANNOTATIONS_TO_COPY = List.of("javax.annotation.security");
```
Could you make it `RolesAllowed.class.getPackageName()`. I'd like to make sure it's rewritten by the Jakarta scripts.

Sgitario commented:
> @Sgitario Hi Jose, can you please add a small section to the main `security.adoc` to link back to the rest-panache doc, for example, just next to https://quarkus.io/guides/security#secure-serialization

Added

Sgitario committed:
Propagate the javax.annotation.security annotations in REST Data

With these changes, the REST Data with Panache extension will propagate the Security annotations within the package `javax.annotation.security` that are defined on your resource interfaces:

```java
import javax.annotation.security.DenyAll;
import javax.annotation.security.RolesAllowed;

@DenyAll
@ResourceProperties
public interface PeopleResource extends PanacheEntityResource<Person, Long> {
    @RolesAllowed("superuser")
    boolean delete(Long id);
}
```

Additionally, if you are only interested in specifying the roles that are allowed to use the resources, the `@ResourceProperties` and `@MethodProperties` annotations have the field `rolesAllowed` to list the security roles permitted to access the resource or operation.

Fix https://github.com/quarkusio/quarkus/issues/28995

sberyozkin commented:
Thanks @Sgitario, we have various security features supported across different extensions, so it makes sense to have a map of all of them in one place :-)

punkratz312 commented:
thanks for fast response
